qiskit_experiments/quantum_volume/qv_analysis.py

```diff
@@ -0,0 +1,282 @@
+        """
+        calculate the probability of measuring heavy output string in the data
+        Args:
+            data (dict): the result of the circuit exectution
+            heavy_outputs (list): the bit strings of the heavy output from the ideal simulation
+        Returns:
+            int: heavy output probability
+        """
+        circ_shots = sum(data["counts"].values())
+
+        # calculate the number of heavy output counts in the experiment
+        heavy_output_counts = sum([data["counts"].get(value, 0) for value in heavy_outputs])
+
+        # calculate the experimental heavy output probability
+        return heavy_output_counts / circ_shots
+
+    @staticmethod
+    def _calc_z_value(mean, sigma):
+        """Calculate z value using mean and sigma.
+
+        Args:
+            mean (float): mean
+            sigma (float): standard deviation
+
+        Returns:
+            float: z_value in standard normal distibution.
+        """
+
+        if sigma == 0:
+            # assign a small value for sigma if sigma = 0
+            sigma = 1e-10
+            warnings.warn("Standard deviation sigma should not be zero.")
+
+        z_value = (mean - 2 / 3) / sigma
+
+        return z_value
+
+    @staticmethod
+    def _calc_confidence_level(z_value):
+        """Calculate confidence level using z value.
+
+        Accumulative probability for standard normal distribution
+        in [-z, +infinity] is 1/2 (1 + erf(z/sqrt(2))),
+        where z = (X - mu)/sigma = (hmean - 2/3)/sigma
+
+        Args:
+            z_value (float): z value in in standard normal distibution.
+
+        Returns:
+            float: confidence level in decimal (not percentage).
+        """
+
+        confidence_level = 0.5 * (1 + math.erf(z_value / 2 ** 0.5))
+
+        return confidence_level
+
+    def _calc_quantum_volume(self, heavy_output_prob_exp, depth, trials):
+        """
+        calc the quantum volume of the analysed system.
+        quantum volume is determined by the largest successful depth.
+        A depth is successful if it has 'mean heavy-output probability' > 2/3 with confidence
+        level > 0.977 (corresponding to z_value = 2).
+        we assume the error (standard deviation) of the heavy output probability is due to a
+        binomial distribution. standard deviation for binomial distribution is sqrt(np(1-p)),
+        where n is the number of trials and p is the success probability.
+
+        Returns:
+            dict: quantum volume calculations -
+            the quantum volume,
+            whether the results passed the threshold,
+            the confidence fof the result,
+            the heavy output probability for each trial,
+            the mean heavy output probability,
+            the error of the heavy output probability,
+            the depth of the circuit,
+            the number of trials ran
+        """
+        quantum_volume = 0
```

Comment:
what does quantum_volume 0 means? (it also appears in the notebook)
it's confusing since 0 is not a power of 2. Maybe default QV value should be 1?